fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -21,27 +21,41 @@
 import java.time.Instant;
 
 import java.util.List;
-import java.util.Set;
 import java.util.stream.Stream;
 
-import org.apache.jena.rdf.model.Resource;
-
 import org.fcrepo.kernel.api.RdfStream;
-import org.fcrepo.kernel.api.TripleCategory;
-import org.fcrepo.kernel.api.identifiers.IdentifierConverter;
 
 /**
+ * A resource in a Fedora repository.
+ *
  * @author ajs6f
  * @since Jan 10, 2014
  */
 public interface FedoraResource {
 
+    /**
+     * Get the fedora identifier for this resource
+     *
+     * @return the fedora identifier
+     */
+    String getId();
+
     /**
      * Get the path to the resource
+     *
      * @return path
+     * @deprecated use getId()
      */
+    @Deprecated
     String getPath();
 
+    /**
+     * Get an object containing header information describing this resource.
+     *
+     * @return header information
+     */
+    ResourceHeaders getHeaders();
```

Comment:
I wonder if we want to ack these as Headers this deep in the system?  That's more the API layer.  But maybe that's okay?

Reply:
I think it's okay for now.  I say let's move forward with the PR.

Reply:
@bseeger :  the reason I say this is that the FedoraResource mediates between the persistence layer and the http api.  So it really is not as deep as you may be thinking.

Reply:
@dbernstein they are coming out of the persistence layer in this PR:
https://github.com/fcrepo4/fcrepo4/pull/1557/files#diff-92d44c0c3bc6cfbd223e78ccc6151bacR62

Although I could see that coming from a kernel layer index client that doesn't currently exist.